src/marqo/s2_inference/multimodal_model_load.py

```diff
@@ -143,13 +145,19 @@ def infer_modality(content: Union[str, List[str], bytes]) -> Modality:
     if isinstance(content, str):
         if not validate_url(content):
             return Modality.TEXT
+        
+        # Encode the URL
+        encoded_url = quote(content, safe=':/')
```

Comment:
Use the function here: https://github.com/marqo-ai/marqo/blob/0f09c36f2f2ab0add8cbb07ce939000fffb932d3/src/marqo/core/inference/image_download.py#L218

Reply:
done